router.lua

```diff
@@ -120,8 +120,18 @@ end
 
 function Router:match(method, path, fun)
   if type(method) == 'string' then -- always make the method to table.
-    method = {[method] = {[path] = fun}}
+    method = {method}
   end
+
+  for i, m in ipairs(method) do
+    -- convert shorthand methods into longhand
+    if type(m) == 'string' then
+      if method[m] == nil then method[m] = {} end
```

Comment:
This is going to be mutating the table. 
I think it would be nicer to just do the mutation in some other table.

For example like:

```lua
local methods = {}
for _, http_method in ipairs(method) 
  if not methods[http_method] then
    methods[http_method] = { [path] = fn }
  else
    methods[http_method][path] = fn
  end
end
```

Reply:
Oh yeah, it is better that way. Let me fix it...